bigframes/session/clients.py

```diff
@@ -104,12 +101,12 @@ def __init__(
             not in bigframes.constants.REP_ENABLED_BIGQUERY_LOCATIONS
         ):
             msg = bfe.format_message(
-                bigframes.constants.LEP_DEPRECATION_WARNING_MESSAGE.format(
+                bigframes.constants.REP_NOT_SUPPOERTED_WARNING_MESSAGE.format(
                     location=location
                 ),
                 fill=False,
             )
-            warnings.warn(msg, category=FutureWarning)
+            warnings.warn(msg, category=ResourceWarning)
```

Comment:
This should be an exception, not a warning.

If we want to make sure that we support new regions as they roll out, the alternative is to warn that this might not work but still use the regional endpoint and let the server side do the failing. I'd prefer that alternative, actually.

Reply:
Resolving. I've implemented the alternative. Please test that it works.